src/control/cmd/daos/util.go

```diff
@@ -138,24 +137,22 @@ func freeString(str *C.char) {
 	C.free(unsafe.Pointer(str))
 }
 
-func createWriteStream(ctx context.Context, prefix string, printLn func(line string)) (*C.FILE, func(), error) {
-	// Create a FILE object for the handler to use for
-	// printing output or errors, and call the callback
-	// for each line.
+func createWriteStream(ctx context.Context, printLn func(line string)) (*C.FILE, func(), error) {
+	// Create a FILE object for the handler to use for printing output or errors, and call the
+	// callback for each line.
 	r, w, err := os.Pipe()
 	if err != nil {
 		return nil, nil, err
 	}
+	done := make(chan bool)
```

Comment:
So if you want to use a channel as a mutex, use a buffered struct{} one: `make(chan struct{}, 1)` as it is memory efficient and deterministic.

Reply:
I'm not sure of the implication of that but I don't want buffering at this point.

Reply:
Ok, I'm reading up on this now.  It looks like this code as-is the write will not happen until the read is called so this could potentially deadlock if the fd is closed for any other reason and I should make the buffer at least one deep in case EOF is seen before the teardown function is called.